### ISSUE TITLE
Carbon::hasFormat("01/01/2019", 'd/m/Y'): preg_match(): Unknown modifier '('

### ISSUE THREAD

**Issue by marktopper** [briannesbitt/Carbon#1886]
Hello,

I encountered an issue with the following code:
```php
Carbon::hasFormat("01/01/2019", 'd/m/Y')
```

Carbon version: 2.24.0

PHP version: 7.2.22

I expected to get:

```
true
```

But I actually get:

```
PHP Warning:  preg_match(): Unknown modifier '(' in /var/www/application/vendor/nesbot/carbon/src/Carbon/Traits/Comparison.php on line 868
```

Thanks!

**Comment by kylekatarnls** [comment]
Fixed, you can try [2.25.0-beta.2](https://github.com/briannesbitt/Carbon/releases/tag/2.25.0-beta.2)

**Comment by devvmiller** [comment]
Thanks guys !
